apps/admin/frontend/src/components/print_diagnostic_button.tsx

```diff
@@ -0,0 +1,134 @@
+import { Button, Modal, P, RadioGroup } from '@votingworks/ui';
+import React, { useState } from 'react';
+import { assert, assertFalsy, sleep } from '@votingworks/basics';
+import type { DiagnosticsOutcome } from '@votingworks/admin-backend';
+import { addDiagnosticRecord, getPrinterStatus, printTestPage } from '../api';
+import { Loading } from './loading';
+
+/**
+ * Normally we want to free up the UI sooner, before the print is actually
+ * done, but in this case we want to wait until the print is done or almost
+ * done before presenting the verification step.
+ */
+export const TEST_PAGE_PRINT_DELAY_SECONDS = 7;
+
+type FlowState = 'printing' | 'verification' | 'test-failed';
+
+export function PrintTestPageButton(): JSX.Element {
+  const [flowState, setFlowState] = useState<FlowState>();
+  const [outcome, setOutcome] = useState<DiagnosticsOutcome>();
+
+  const printerStatusQuery = getPrinterStatus.useQuery();
+  const printTestPageMutation = printTestPage.useMutation();
+  const addDiagnosticRecordMutation = addDiagnosticRecord.useMutation();
+
+  const isPrinterConnected =
+    printerStatusQuery.isSuccess && printerStatusQuery.data.connected;
+
+  function resetFlow() {
+    setFlowState(undefined);
+    setOutcome(undefined);
+    printTestPageMutation.reset();
+    addDiagnosticRecordMutation.reset();
```

Comment:
Not necessary. I think it is nice to have mutations mount in the modal (not the button) so whenever you start the flow again the mutations reset. So when I decided to just put all the state in a single button component for other reasons, I thought I would reset the mutations on close in case we ever use state from them. But I guess that is overthinking it and it looks like their presence is more confusing, so will remove.